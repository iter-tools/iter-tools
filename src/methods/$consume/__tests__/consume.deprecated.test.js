/**
 * @generated-from ./$consume.deprecated.test.js
 * This file is autogenerated from a template. Please do not edit it directly.
 * To rebuild it from its template use the command
 * > npm run generate
 * More information can be found in CONTRIBUTING.md
 */

/* eslint-disable no-unused-vars,import/no-duplicates,no-constant-condition */

import { consume } from '../../..';
describe('consume', () => {
  it('consumes an iterable with a callback', () => {
    const arr: Array<number> = [];
    consume(item => arr.push(item), [1, 2, 3]);
    expect(arr).toEqual([1, 2, 3]);
  });
  it('consumes an iterable with a callback returning a promise', () => {
    const arr: Array<number> = [];
    consume(
      item => {
        arr.push(item);
        return Promise.resolve(0);
      },
      [1, 2, 3],
    );
    expect(arr).toEqual([1, 2, 3]);
  });
  it('consumes an iterable (curried)', () => {
    const arr: Array<number> = [];
    const consumePush = consume((item: number) => arr.push(item));
    consumePush([1, 2, 3]);
    expect(arr).toEqual([1, 2, 3]);
  });
});
