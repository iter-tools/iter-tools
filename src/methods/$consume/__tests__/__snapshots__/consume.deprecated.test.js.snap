// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`consume (deprecated) consumes an iterable with a callback 1`] = `
Array [
  Array [
    "\`consume(callback, iterable)\` is deprecated and will be removed in iter-tools@8. Instead use forEach(callback, iterable)",
  ],
]
`;
